src/executor.tsx

```diff
@@ -64,12 +67,31 @@ export async function execute({
     return Map();
   }
   const { user } = plan;
-  if (!isUserLoggedIn(user)) {
+
+  const isLoggedIn = isUserLoggedIn(user) || isUserLoggedInWithExtension(user);
+  if (!isLoggedIn) {
     return Map();
   }
-  const finalizedEvents = plan.publishEvents.map((e) =>
-    finalizeEvent(e, user.privateKey)
-  );
+
+  const signEventWithExtension = async (
+    event: EventTemplate
+  ): Promise<Event> => {
+    try {
+      return window.nostr.signEvent(event);
+      // eslint-disable-next-line no-empty
+    } catch {
+      throw new Error("Failed to sign event with extension");
+    }
+  };
+  const finalizedEvents = isUserLoggedInWithExtension(user)
```

Comment:
I think it would be better if the `NostrProvider` who provides `finalizeEvent` deals with that. The executor is the wrong place.

Reply:
Unluckily, `NostrProvider` lives without login information from `NostrAuthContext`; so it can't handle the signing/finalizing depending on the login-method. 
Despite, finalizeEvent only finalizes one event, so the executor coordinates the finalization of multiple events now anyways..